boards/nrf51dk/src/main.rs

```diff
@@ -38,6 +38,8 @@
 #![no_std]
 #![no_main]
 #![feature(lang_items, compiler_builtins_lib)]
+#![deny(missing_docs)]
+#![deny(warnings)]
```

Comment:
I don't think we want to deny warnings here, if I understand correctly, that'll cause all local compiles with warnings to fail, which is a bit obnoxious during development. If everything is set up correctly, Travis should reject builds with warnings so we won't merge them, but you don't have to be warnings-perfect while you develop. I think this is the right balance.

Reply:
Yepp, that’s my understanding as well but yeah I agree better to leave warnings  to Travis. But currently I think warnings in Rust code is not regarded as fail in Travis right?

BTW, I think we should strive to require docs in all our crates but this is just a start.